Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs

```diff
@@ -16,529 +16,632 @@
 
 namespace Magick.NET.Tests
 {
-    [TestClass]
-    public class ImageOptimizerTests : ImageOptimizerTestHelper
+    public partial class ImageOptimizerTests
     {
-        private ImageOptimizer Optimizer => new ImageOptimizer();
-
-        [TestMethod]
-        public void Compress_FileIsNull_ThrowsException()
-        {
-            ExceptionAssert.ThrowsArgumentNullException("file", () =>
-            {
-                Optimizer.Compress((FileInfo)null);
-            });
-        }
-
-        [TestMethod]
-        public void Compress_FileNameIsNull_ThrowsException()
-        {
-            ExceptionAssert.ThrowsArgumentNullException("fileName", () =>
-            {
-                Optimizer.Compress((string)null);
-            });
-        }
-
-        [TestMethod]
-        public void Compress_FileNameIsEmpty_ThrowsException()
-        {
-            ExceptionAssert.ThrowsArgumentException("fileName", () =>
-            {
-                Optimizer.Compress(string.Empty);
-            });
-        }
-
-        [TestMethod]
-        public void Compress_FileNameIsInvalid_ThrowsException()
-        {
-            ExceptionAssert.Throws<MagickBlobErrorException>(() =>
-            {
-                Optimizer.Compress(Files.Missing);
-            }, "error/blob.c/OpenBlob");
-        }
-
-        [TestMethod]
-        public void Compress_InvalidFile_ThrowsException()
+        private static FileStream OpenFile(string path)
         {
-            ExceptionAssert.Throws<MagickCorruptImageErrorException>(() =>
-            {
-                Optimizer.Compress(Files.InvitationTif);
-            }, "Invalid format");
+            return File.Open(path, FileMode.Open, FileAccess.ReadWrite);
         }
 
-        [TestMethod]
-        public void Compress_EmptyFile_ThrowsException()
+        [TestClass]
+        public class TheCompressMethod
```

Comment:
Each class should be in it's own file inside a folder with the name of the class + `Tests`. This one should be inside `Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests/TheCompressMethod.cs` and the rest should follow the same pattern. This class should also inherit from `ImageOptimizerTests`. The shared logic (`static FileStream OpenFile`) should be inside `Init.cs`. You can find an example in the `ExifProfileTests` folder.

Reply:
Sorry about that, now the class for each method is in its own file and the shared logic has been moved to `Init.cs`.